File: src/leetcode/python/ExcelSheetColumnNumber_171.py
class ExcelSheetColumnNumber_171(object):
    def titleToNumber(self, s):
        """
        :type s: str
        :rtype: int
        """
        res, k = 0, 1
        s = s[::-1]
        for c in s:
            res += (ord(c) - 64) * k
            k *= 26
        return res

File: src/leetcode/java/ExcelSheetColumnNumber_171.java
package leetcode.java;

public class ExcelSheetColumnNumber_171 {
  public int titleToNumber(String s) {
    char[] c = s.toCharArray();
    int res = 0;
    for (int i = c.length - 1, k = 1; i >= 0; i--, k *= 26) {
      res += (c[i] - 'A' + 1) * k;
    }
    return res;
  }
}
